File: mobile/lib/core/models/vpn_protocol.dart
enum VpnProtocol {
  wireguard('wireguard'),
  vless('vless');

  const VpnProtocol(this.value);
  final String value;

  String get label {
    switch (this) {
      case VpnProtocol.vless:
        return 'VLESS';
      case VpnProtocol.wireguard:
        return 'WireGuard';
    }
  }

  static VpnProtocol fromString(String? raw) {
    switch (raw?.toLowerCase()) {
      case 'vless':
        return VpnProtocol.vless;
      case 'wireguard':
      default:
        return VpnProtocol.wireguard;
    }
  }
}

File: backend/app/Enums/VpnProtocol.php
<?php

namespace App\Enums;

enum VpnProtocol: string
{
    case Wireguard = 'wireguard';
    case Vless = 'vless';

    public function peerCodePrefix(): string
    {
        return match ($this) {
            self::Wireguard => 'WG-PEER-',
            self::Vless => 'VLESS-PEER-',
        };
    }

    public function requiresIpAllocation(): bool
    {
        return $this === self::Wireguard;
    }

    /**
     * @return list<string>
     */
    public static function values(): array
    {
        return array_column(self::cases(), 'value');
    }
}
